src/DSharp.Script/src/System/Collections/CollectionHelpers.js

```diff
@@ -5,7 +5,23 @@ function toArray(obj) {
             : Array.prototype.slice.call(obj)
         : null;
 }
+function removeAt(a, index) {
+    // custom implementation
+    if (typeof a.removeAt === "function") {
```

Comment:
We could make this a generic function which can generate this pattern as we use it quite a lot. 
Thinking something like this:
```javascript
function createFallbackFunction(name, fallback){
    return function(instance){
        if(typeof instance[name] === "function"){
            return instance[name].apply(null, arguments);
        }

        return fallback.apply(null, arguments);
    }
}
```
Then you define the `removeAt` method like:
```javascript
var removeAt = createFallbackFunction("removeAt", function(a, index){
    return index >= 0
        ? (a.splice(index, 1), true)
        : false;
});
```

Reply:
This would be out of the scope of the PR anyhow, but worth mentioning as we generally try to keep the SS script as small as possible :)